src/main/scala/amba/axis/Parameters.scala

```diff
@@ -0,0 +1,93 @@
+// See LICENSE.SiFive for license details.
+package freechips.rocketchip.amba.axis
+
+import chisel3._
+import chisel3.util._
+import chisel3.internal.sourceinfo.SourceInfo
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.util.BundleField
+import freechips.rocketchip.diplomacy._
+
+case class AXISSlaveParameters(
+  name:          String,
+  destinationId: Int           = 0,
+  resources:     Seq[Resource] = Nil,
+  nodePath:      Seq[BaseNode] = Nil,
+  supportsWrite: TransferSizes = TransferSizes.none,
+  supportsRead:  TransferSizes = TransferSizes.none)
+
+case class AXISSlavePortParameters(
+  slaves:    Seq[AXISSlaveParameters],
+  beatBytes: Option[Int] = None)
+{
+  require (!slaves.isEmpty)
+  beatBytes.foreach { b => require(isPow2(b)) }
+  val endDestinationId = slaves.map(_.destinationId).max + 1
+}
+
+case class AXISMasterParameters(
+  name:     String,
+  sourceId: IdRange       = IdRange(0,1),
+  nodePath: Seq[BaseNode] = Nil)
+{
+  require (!sourceId.isEmpty)
+}
+
+case class AXISMasterPortParameters(
+  masters:    Seq[AXISMasterParameters],
+  userFields: Seq[BundleField] = Nil,
+  beatBytes:  Option[Int]      = None)
+{
+  require (!masters.isEmpty)
+  beatBytes.foreach { b => require(isPow2(b)) }
+  val endSourceId = masters.map(_.sourceId.end).max
+}
+
+case class AXISBundleParameters(
+  idBits:     Int,
+  destBits:   Int,
+  dataBits:   Int,
+  userFields: Seq[BundleField])
+{
+  require (idBits >= 0)
+  require (destBits >= 0)
+  require (dataBits >= 8)
+  require (isPow2(dataBits))
+
+  val keepBits = dataBits/8
+  val strbBits = dataBits/8
+
+  def union(x: AXISBundleParameters) = AXISBundleParameters(
+    idBits     = idBits   max x.idBits,
+    destBits   = destBits max x.destBits,
+    dataBits   = dataBits max x.dataBits,
+    userFields = (userFields ++ x.userFields).distinct)
+}
+
+object AXISBundleParameters {
+  val emptyBundleParams = AXISBundleParameters(
+    idBits     = 0,
+    destBits   = 0,
+    dataBits   = 8,
+    userFields = Nil)
+  def union(x: Seq[AXISBundleParameters]) = x.foldLeft(emptyBundleParams)((x,y) => x.union(y))
+}
+
+case class AXISEdgeParameters(
+  master:     AXISMasterPortParameters,
+  slave:      AXISSlavePortParameters,
+  params:     Parameters,
+  sourceInfo: SourceInfo)
+{
+  require (slave.beatBytes.isEmpty && master.beatBytes.isEmpty,
+    s"Neither master nor slave port specify a bus width (insert an AXISBusBinder between them?) at ${sourceInfo}")
+  require (!slave.beatBytes.isEmpty && !master.beatBytes.isEmpty && slave.beatBytes != master.beatBytes,
+    s"Master and slave ports specify incompatible bus widths (insert an AXISWidthWidget between them?) at ${sourceInfo}")
```

Comment:
Are the BusBinders and WidthWidgets coming in later PRs?

Reply:
Yes